src/ranged.cpp

```diff
@@ -1219,6 +1219,21 @@ void practice_archery_proficiency( player &p, const item &relevant )
     }
 }
 
+// Apply stamina cost to archery which decreases due to proficiency
+void mod_stamina_archery( player &p )
+{
+    // Set activity level if it's below that level
+    if( p.activity_history.activity() < ACTIVE_EXERCISE ) {
+        p.set_activity_level( ACTIVE_EXERCISE );
+    }
```

Comment:
```suggestion
    p.set_activity_level( ACTIVE_EXERCISE );
```
The check isn't needed; it always takes the highest value happening that turn.

Reply:
I'm trying to avoid excessive calculation since do_aim() occurs on a per-move basis (100 times per turn). Should that check still be dropped anyway?

Reply:
Why would calling p.activity_history.activity() result in significantly less overhead than calling p.set_activity_level? And branching isn't great for performance.

Reply:
I actually never knew branching was bad for performance, I had to look up why